app/main/helpers/validation_tools.py

```diff
@@ -16,7 +21,8 @@ def __init__(self, content, service, posted_data, uploader=None):
         self.service = service
         self.posted_data = posted_data
         self.uploader = uploader
-        self.clean_data = None
```

Comment:
Setting `_data` fields to `None` initially makes dictionary operations fail on them, which means doing something like `service.update(clean_data)` without calling `.errors` first is an explicit error.